libs/prelude/Prelude/Types.idr

```diff
@@ -390,6 +393,12 @@ Foldable List where
 
   toList = id
 
+  foldMap f xs = go xs neutral
+    where
+      go : List a -> m -> m
+      go [] z = z
+      go (x :: xs) z = go xs (z <+> f x)
+
```

Comment:
Why not just `foldMap f = foldl (\acc, elem => acc <+> f elem) neutral` as soon as `foldl` is already tail-recursive?